log_pipeline_event.py

```diff
@@ -0,0 +1,53 @@
+import argparse
+import json
+
+from core_data_modules.logging import Logger
+from storage.google_cloud import google_cloud_utils
+from pipeline_logs.firestore_pipeline_logger import FirestorePipelineLogger
+
+from src.lib import PipelineConfiguration
+from src.lib.configuration_objects import PipelineEvents
+
+log = Logger(__name__)
+
+def log_pipeline_event(user, google_cloud_credentials_file_path, pipeline_configuration_file_path, timestamp, run_id, event_key):
+    # Read the settings from the configuration file
+    log.info("Loading Pipeline Configuration File...")
+    with open(pipeline_configuration_file_path) as f:
+        pipeline_configuration = PipelineConfiguration.from_configuration_file(f)
+
+    log.info("Downloading Firestore Operations Dashboard Table credentials...")
+    firestore_pipeline_logs_table_credentials = json.loads(google_cloud_utils.download_blob_to_string(
+        google_cloud_credentials_file_path,
+        pipeline_configuration.operations_dashboard.firebase_credentials_file_url
+    ))
+
+    log.info(f"Updating PipelineStart event log for run_id: {run_id}")
+    firestore_pipeline_logger= FirestorePipelineLogger( pipeline_configuration.pipeline_name, timestamp, run_id,
+                                                  PipelineEvents.PIPELINE_EVENT[event_key], firestore_pipeline_logs_table_credentials)
+
+    firestore_pipeline_logger.update_pipeline_logs()
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Updates current pipeline event/stage to a firebase table to aid in monitoring")
+
+    parser.add_argument("user", help="Identifier of the user launching this program")
+    parser.add_argument("google_cloud_credentials_file_path", metavar="google-cloud-credentials-file-path",
+                        help="Path to a Google Cloud service account credentials file to use to access the "
+                             "credentials bucket")
+    parser.add_argument("pipeline_configuration_file_path", metavar="pipeline-configuration-file",
+                        help="Path to the pipeline configuration json file"),
+    parser.add_argument("timestamp", metavar="timestamp",
+                        help="current pipeline run start datetime")
+    parser.add_argument("run_id", metavar="run-id",
+                        help="Identifier of this pipeline run")
+    parser.add_argument("event_key", metavar="event-key",
+                        help="Key for this pipeline event/stage {1:CodaAdd, 2:FetchingRawData, 3:GeneratingOutputs, 4:CodaAdd,\
+                              5:GeneratingAutomatedAnalysisFiles, 6:BackingUpData, 7:UploadingAnalysisFiles,\
+                              8:UploadingLogFiles, 9:PipelineCompletedSuccessfully}",
+                              choices= ["1","2","3","4","5","6","7","8","9"]),
```

Comment:
Can the choices be the available string options rather than numbers? i.e.
```suggestion
                              choices= ["PipelineStart", "CodaAdd", ...]),
```
That way I don't need to look up the mapping when I'm reading the run scripts.
Also I think PipelineStart should be an option.